chia/full_node/block_store.py

```diff
@@ -50,19 +50,21 @@ async def create(cls, db_wrapper: DBWrapper):
 
         # Height index so we can look up in order of height for sync purposes
         await self.db.execute("CREATE INDEX IF NOT EXISTS full_block_height on full_blocks(height)")
-        # this index is not used by any queries, don't create it for new
-        # installs, and remove it from existing installs in the future
-        # await self.db.execute("DROP INDEX IF EXISTS is_block on full_blocks(is_block)")
         await self.db.execute("CREATE INDEX IF NOT EXISTS is_fully_compactified on full_blocks(is_fully_compactified)")
 
         await self.db.execute("CREATE INDEX IF NOT EXISTS height on block_records(height)")
 
         await self.db.execute("CREATE INDEX IF NOT EXISTS hh on block_records(header_hash)")
-        await self.db.execute("CREATE INDEX IF NOT EXISTS peak on block_records(is_peak)")
 
-        # this index is not used by any queries, don't create it for new
-        # installs, and remove it from existing installs in the future
-        # await self.db.execute("DROP INDEX IF EXISTS is_block on block_records(is_block)")
+        if self.db.allow_upgrades:
+            await self.db.execute("DROP INDEX IF EXISTS is_block on full_blocks(is_block)")
+            await self.db.execute("DROP INDEX IF EXISTS is_block on block_records(is_block)")
+            await self.db.execute("DROP INDEX IF EXISTS peak")
+            await self.db.execute(
+                "CREATE INDEX IF NOT EXISTS is_peak_eq_1_idx on block_records(is_peak) where is_peak = 1"
+            )
+        else:
+            await self.db.execute("CREATE INDEX IF NOT EXISTS peak on block_records(is_peak) where is_peak = 1")
```

Comment:
both new users, as well as users that don't opt-in to `allow_upgrades` will take this path. That's why it's important this still uses the old name for the index, to make it a no-op for existing users. However, for new users, this old name will now use the more compact index.

Reply:
Why not force users to do the upgrade? That way we can guarantee that everyone running the latest version will have the DB upgrades, instead of getting split users with and without the upgrade.

Is it very very slow? Perhaps we can show a loading screen

Reply:
it can be very slow, yes. If you (like myself) has the database on a spinning disk. However, I don't have a strong opinion, I think it's fine to incur this one-time cost on restart. Bill suggested we be a bit more conservative though, and recommend people that already are having a hard time keeping up to enable this